### PR TITLE
Add effects tab to vehicles

### PR DESCRIPTION
**Pull request by LebombJames**
Known issue: conditions don't roll open with their description on click even though it should've been inherited from the base class. Might be an intended feature of inactive conditions?

### CODE REVIEW

**Comment by CarlosFdez** [comment]
Condition summary viewing is bugged for other sheet types as well currently.

File: src/module/actor/vehicle/sheet.ts [hunk 1]
```diff
@@ -86,5 +88,35 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
 
         // get buttons
         $html.find(".crb-tag-selector").on("click", (event) => this.onTraitSelector(event));
+
+        // Change whether an effect is secret to players or not
+        $html.find(".effects-list [data-action=effect-toggle-unidentified]").on("click", async (event) => {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:93]: no jqueriez

Comment (LebombJames) [reply src/module/actor/vehicle/sheet.ts:93]: I copied this from the PC sheet. Is this just a case of not adding *more* jQuery if it can be helped?

**Comment by LebombJames** [comment]
Required adjustment to the types taken on traitSelector to support both jQuery events and vanilla JS events.
Un/identification is on `creature`, it's just that `vehicle` doesn't extend `creature`, rather `actor` directly, so it doesn't inherit it.

File: src/module/actor/vehicle/sheet.ts [hunk 2]
```diff
@@ -70,29 +72,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
     override activateListeners($html: JQuery): void {
         super.activateListeners($html);
+        const html = $html[0];
         {
             // ensure correct tab name is displayed after actor update
-            const title = $(".sheet-navigation .active").attr("title");
-            if (title) {
-                $html.find(".navigation-title").text(title);
-            }
+            const title = html.querySelector(".sheet-navigation .active")?.getAttribute("title");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:78]: it's just `.title`

File: src/module/actor/vehicle/sheet.ts [hunk 3]
```diff
@@ -70,29 +72,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
     override activateListeners($html: JQuery): void {
         super.activateListeners($html);
+        const html = $html[0];
         {
             // ensure correct tab name is displayed after actor update
-            const title = $(".sheet-navigation .active").attr("title");
-            if (title) {
-                $html.find(".navigation-title").text(title);
-            }
+            const title = (html.querySelector(".sheet-navigation .active") as HTMLElement)?.title;
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:78]: `htmlQuery`

File: src/module/actor/vehicle/sheet.ts [hunk 4]
```diff
@@ -1,7 +1,9 @@
 import { ActorSheetPF2e } from "../sheet/base";
 import { VehiclePF2e } from "@actor/vehicle";
 import { ItemDataPF2e } from "@item/data";
-import { getActionIcon } from "@util";
+import { getActionIcon, htmlQuery, htmlQueryAll } from "@util";
+import { htmlClosest } from "@util";
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:5]:
```suggestion
import { getActionIcon, htmlClosest, htmlQuery, htmlQueryAll } from "@util";
```

File: src/module/actor/vehicle/sheet.ts [hunk 5]
```diff
@@ -70,29 +71,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
-            const parent = $(event.currentTarget).parents(".sheet-navigation");
-            const title = parent.find(".item.active").attr("title");
-            if (title) {
-                parent.find(".navigation-title").text(title);
-            }
-        });
 
         // get buttons
-        $html.find(".crb-tag-selector").on("click", (event) => this.onTraitSelector(event));
+        for (const element of htmlQueryAll(html, ".crb-tag-selector") ?? []) {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:104]:
```suggestion
        for (const element of htmlQueryAll(html, ".crb-tag-selector")) {
```

File: src/module/actor/vehicle/sheet.ts [hunk 6]
```diff
@@ -70,29 +71,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                const effect = this.actor.items.get(effectId, { strict: true });
+                if (effect instanceof AbstractEffectPF2e) {
+                    const isUnidentified = effect.unidentified;
+                    await effect.update({ "system.unidentified": !isUnidentified });
+                }
+            });
+        }
+
+        // Decrease effect value
+        for (const element of htmlQueryAll(html, ".effects-list .decrement") ?? []) {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:121]:
```suggestion
        for (const element of htmlQueryAll(html, ".effects-list .decrement")) {
```

File: src/module/actor/vehicle/sheet.ts [hunk 7]
```diff
@@ -70,29 +71,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                const parent = htmlClosest(el.parentElement, ".item");
+                const effect = this.actor.items.get(parent?.dataset.dataItemId ?? "");
+                if (effect instanceof AbstractEffectPF2e) {
+                    await effect.decrease();
+                }
+            });
+        }
+
+        // Increase effect value
+        for (const element of htmlQueryAll(html, ".effects-list .increment") ?? []) {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:133]:
```suggestion
        for (const element of htmlQueryAll(html, ".effects-list .increment")) {
```

File: src/module/actor/vehicle/sheet.ts [hunk 8]
```diff
@@ -70,29 +71,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                    await effect.decrease();
+                }
+            });
+        }
+
+        // Increase effect value
+        for (const element of htmlQueryAll(html, ".effects-list .increment")) {
+            element.addEventListener("click", async (event) => {
+                const el = event.currentTarget as HTMLElement;
+                const parent = htmlClosest(el.parentElement, ".item");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:136]:
```suggestion
                const parent = htmlClosest(element.parentElement, ".item");
```

File: src/module/actor/vehicle/sheet.ts [hunk 9]
```diff
@@ -70,29 +71,75 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                    await effect.update({ "system.unidentified": !isUnidentified });
+                }
+            });
+        }
+
+        // Decrease effect value
+        for (const element of htmlQueryAll(html, ".effects-list .decrement")) {
+            element.addEventListener("click", async (event) => {
+                const el = event.currentTarget as HTMLElement;
+                const parent = htmlClosest(el.parentElement, ".item");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:124]:
```suggestion
                const parent = htmlClosest(element.parentElement, ".item");
```

File: src/module/actor/vehicle/sheet.ts [hunk 10]
```diff
@@ -70,29 +71,73 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                const el = event.currentTarget as HTMLElement;
+                const title = el?.title;
+                if (title) {
+                    const parent = htmlQuery(el.parentElement, ".navigation-title");
+                    if (parent) parent.textContent = title;
+                }
+            });
+        }
+
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item") ?? []) {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:91]:
```suggestion
        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
            element.addEventListener("mouseover", (event) => {
                    const parent = htmlQuery(el.parentElement, ".navigation-title");
                    parent.textContent = element.title;
                }
            });
        }

        for (const element of htmlQueryAll(html, ".sheet-navigation .item") {
```

File: src/module/actor/vehicle/sheet.ts [hunk 11]
```diff
@@ -70,29 +71,70 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
-        });
 
         // get buttons
-        $html.find(".crb-tag-selector").on("click", (event) => this.onTraitSelector(event));
+        for (const element of htmlQueryAll(html, ".crb-tag-selector")) {
+            element.addEventListener("click", (event: MouseEvent) => this.onTraitSelector(event));
+        }
+
+        // Change whether an effect is secret to players or not
+        for (const element of htmlQueryAll(html, ".effects-list [data-action=effect-toggle-unidentified]") ?? []) {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:106]:
```suggestion
        for (const element of htmlQueryAll(html, ".effects-list [data-action=effect-toggle-unidentified]")) {
```

File: src/module/actor/vehicle/sheet.ts [hunk 12]
```diff
@@ -70,29 +71,70 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+            element.addEventListener("click", async (event) => {
+                const effectId = htmlClosest(event.currentTarget, "[data-item-id]")?.dataset.itemId;
+                const effect = this.actor.items.get(effectId, { strict: true });
+                if (effect instanceof AbstractEffectPF2e) {
+                    const isUnidentified = effect.unidentified;
+                    await effect.update({ "system.unidentified": !isUnidentified });
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:112]: Conditions shouldn't be markable as unidentified

File: src/module/actor/vehicle/sheet.ts [hunk 13]
```diff
@@ -70,29 +71,70 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                    await effect.decrease();
+                }
+            });
+        }
+
+        // Increase effect value
+        for (const element of htmlQueryAll(html, ".effects-list .increment")) {
+            element.addEventListener("click", async (event) => {
+                const parent = htmlClosest(element.parentElement, ".item");
+                const effect = this.actor.items.get(parent?.dataset.dataItemId ?? "");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:132]:
```suggestion
                const effect = this.actor?.items.get(parent?.dataset.dataItemId ?? "");
```

**Review by stwlam — DISMISSED** [review]

File: src/module/actor/vehicle/sheet.ts [hunk 14]
```diff
@@ -70,29 +71,70 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+        }
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
+            element.addEventListener("mouseover", (event) => {
+                    const parent = htmlQuery(el.parentElement, ".navigation-title");
+                    parent.textContent = element.title;
+                }
+            });
+        }
+
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item") {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:88]:
```suggestion
        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
```

**Comment by LebombJames** [comment]
Should I change the functions back to using `event`?

File: src/module/actor/vehicle/sheet.ts [hunk 15]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
             // ensure correct tab name is displayed after actor update
-            const title = $(".sheet-navigation .active").attr("title");
-            if (title) {
-                $html.find(".navigation-title").text(title);
-            }
+            const title = htmlQuery(html, ".sheet-navigation .active")?.title;
+            if (title) htmlQuery(html, ".navigation-title")!.textContent = title;
+        }
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
+            element.addEventListener("mouseover", (event) => {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:81]:
```suggestion
            element.addEventListener("mouseover", () => {
```

File: src/module/actor/vehicle/sheet.ts [hunk 16]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                const parent = htmlQuery(element.parentElement, ".navigation-title");
+                parent!.textContent = element.title;
+            });
+        }
+
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
+            element.addEventListener("mouseout", (event) => {
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:88]:
```suggestion
            element.addEventListener("mouseout", () => {
```

File: src/module/actor/vehicle/sheet.ts (hunk 15 above)
Comment (LebombJames) [reply src/module/actor/vehicle/sheet.ts]: Or that

File: src/module/actor/vehicle/sheet.ts [hunk 17]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                const el = event.currentTarget as HTMLElement;
+                const parent = htmlClosest(el.parentElement, ".sheet-navigation");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:90]:
```suggestion
                const parent = htmlClosest(element, ".sheet-navigation");
```

File: src/module/actor/vehicle/sheet.ts [hunk 18]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                    const isUnidentified = effect.unidentified;
+                    await effect.update({ "system.unidentified": !isUnidentified });
+                }
+            });
+        }
+
+        // Decrease effect value
+        for (const element of htmlQueryAll(html, ".effects-list .decrement")) {
+            element.addEventListener("click", async (event) => {
+                const parent = htmlClosest(element.parentElement, ".item");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:119]:
```suggestion
                const parent = htmlClosest(element, ".item");
```

File: src/module/actor/vehicle/sheet.ts [hunk 19]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
+                if (effect instanceof AbstractEffectPF2e) {
+                    await effect.decrease();
+                }
+            });
+        }
+
+        // Increase effect value
+        for (const element of htmlQueryAll(html, ".effects-list .increment")) {
+            element.addEventListener("click", async (event) => {
+                const parent = htmlClosest(element.parentElement, ".item");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:130]:
```suggestion
                const parent = htmlClosest(element, ".item");
```

File: src/module/actor/vehicle/sheet.ts [hunk 20]
```diff
@@ -70,29 +71,69 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
-            const title = $(".sheet-navigation .active").attr("title");
-            if (title) {
-                $html.find(".navigation-title").text(title);
-            }
+            const title = htmlQuery(html, ".sheet-navigation .active")?.title;
+            if (title) htmlQuery(html, ".navigation-title")!.textContent = title;
+        }
+        for (const element of htmlQueryAll(html, ".sheet-navigation .item")) {
+            element.addEventListener("mouseover", (event) => {
+                const parent = htmlQuery(element.parentElement, ".navigation-title");
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:82]:
```suggestion
                const parent = htmlQuery(element, ".navigation-title");
```

File: src/module/actor/vehicle/sheet.ts [hunk 21]
```diff
@@ -70,29 +71,68 @@ export class VehicleSheetPF2e extends ActorSheetPF2e<VehiclePF2e> {
...
         // get buttons
-        $html.find(".crb-tag-selector").on("click", (event) => this.onTraitSelector(event));
+        for (const element of htmlQueryAll(html, ".crb-tag-selector")) {
+            element.addEventListener("click", (event: MouseEvent) => this.onTraitSelector(event));
+        }
+
+        // Change whether an effect is secret to players or not
+        for (const element of htmlQueryAll(html, ".effects-list [data-action=effect-toggle-unidentified]")) {
+            element.addEventListener("click", async (event) => {
+                const effectId = htmlClosest(event.currentTarget, "[data-item-id]")?.dataset.itemId;
```
Comment (stwlam) [inline src/module/actor/vehicle/sheet.ts:106]:
```suggestion
            element.addEventListener("click", async () => {
                const effectId = htmlClosest(element, "[data-item-id]")?.dataset.itemId;
```

**Pull request merged**